azurerm/internal/services/hdinsight/hdinsight.go

```diff
@@ -754,14 +754,19 @@ func SchemaHDInsightNodeDefinition(schemaLocation string, definition HDInsightNo
 		}
 	}
 
-	return &schema.Schema{
+	s := &schema.Schema{
 		Type:     schema.TypeList,
-		Required: true,
 		MaxItems: 1,
```

Comment:
fwiw you could make this:

```suggestion
		Required: required,
		Optional: !required,
		MaxItems: 1,
```

Reply:
Thank you for the hint!